Add support for interfaceId. (#8642)
ethereum/solidity: docs/units-and-global-variables.rst

    regular calls.
    The same restrictions as with ``.creationCode`` also apply for this
    property.

Additionally to the properties that are available for contract types,
a different set is available for interface type ``I``:

``type(I).interfaceId``:

>     This property returns the ``EIP-165`` interface identifier of the given interface ``I``, excluding all

Reviewer: Should link to https://eips.ethereum.org/EIPS/eip-165

Author: ah nice :)

    regular calls.
    The same restrictions as with ``.creationCode`` also apply for this
    property.

> Additionally to the properties that are available for contract types,
> a different set is available for interface type ``I``:

Reviewer 2: ```suggestion
In addition to the properties above, the following properties are available for an interface type ``I``:
```

``type(I).interfaceId``:

>     This property returns the `EIP-165 <https://eips.ethereum.org/EIPS/eip-165>`_

Reviewer 2: Sorry to comment here again: Strictly, only functions return values and this is not a function.
Maybe
`A ``bytes4`` value containing the `EIP-165..`
and then you can remove the last sentence in the paragraph.

>     interface identifier of the given interface ``I``, excluding all
>     potentially derived functions. The interface identifier is defined as the ``XOR`` of all
>     function selectors defined within the interface. The returned interface identifier is of type ``bytes4``.

Reviewer 2: ```suggestion
    interface identifier of the given interface ``I``. This identifier is defined as the ``XOR`` of all
    function selectors defined within the interface itself - excluding all inherited functions. The returned interface identifier is of type ``bytes4``.
```
